### ISSUE TITLE
Review

### ISSUE THREAD

**Issue by Simone-Bordoni** [qiboteam/rl-noisemodel#68]
### Section II.A ###

- [ ] (@AlejandroSopena) The RB noise model could benefit from its own subsection, clarifying on how the decay parameter is used to specify the depolarizing rate for each gate. Remember that for single-qubit circuit where a Clifford gate decomposes to RZ*RX*RZ*RX*RZ (i.e. each Clifford gate uses 2 RX gates), the error-per-Clifford = error-per-RX-gate/2. The formula is slightly more complicated for 2Q and 3Q Clifford gates.

We can consider using an appendix for this. We should also specify that the gates used are natives so they don't require a decomposition.

### Section II.B ###

- [ ] (@BrunoLiegiBastonLiegi) A brief explanation of the Proximal Policy Optimization (PPO) algorithm is necessary. Specifically, the manuscript should describe the objective function (expected cumulative return) and justify why PPO was chosen over other RL algorithms like Q-learning or Soft Actor-Critic.

Also here we should consider using an appendix for the more technical part of RL. Regarding why we have used PPO it is mainly becase it allows continuous action values and works better empirically

### Section III.A ###

- [ ]  (@Simone-Bordoni) What happens at the beginning and end of the circuit with a window size of k=3? Does the model pad the circuit or treat the edges differently?

I will just add that we are padding the edges (I thought we already specified this...)

### Section III.B ###

- [ ] (@Simone-Bordoni) In Equation 5, is there a specific reason why the Trace Distance (TD) is squared?

I will just point out that we have tried different functional forms and this was the best reward empirically

### Section IV.A ###

- [ ] (@BrunoLiegiBastonLiegi) The authors mention that in their case, moving from Clifford to non-Clifford circuits involves allowing RZ rotations beyond pi/2. This point should be reiterated later to clarify the difference between some test sets to the training set.

I don't understand why they want us to write this again but ok...

- [ ] (@AlejandroSopena) Most readers understand RB as a characterization method to extract an average error-per-gate. Therefore, the authors should take steps to clarify that the RB noise model additionally takes the decay parameter as the depolarizing rate for every gate in the circuit.

Maybe we should point out more the difference between the standard RB and how we used this technique to produce a noise model, I think this sould be done in the section dedicated to RB.

- [ ] (@Simone-Bordoni) The claim “The improvement is especially pronounced on shorter circuits” is contradicted by Figure 5. The figure does not show RL getting worse or RB improving for longer circuits.

I will remove this sentence, I think it was related to previous results where this behaviour was more evident.

- [ ] (@Simone-Bordoni) All figure captions should be more descriptive, pointing out key observations for quick understanding.

- [ ] (@Simone-Bordoni) Figure 5: Why is the maximally mixed state case included? Also, better motivation for the noiseless case would help readers understand its relevance.

I will add e few more lines on this.

- [ ] (@Simone-Bordoni) The authors should consider combining Figures 4 and 6 to emphasize differences in convergence times for 1-qubit and 3-qubit cases.

I think we should discuss this (I will make a preview of the combined figures and vote the best option).

- [ ] (@Simone-Bordoni) A table summarizing model performance on different test sets would better highlight generalization (Clifford vs. non-Clifford, varying depths, etc.).

I can make this table however I am not sure how to put the results on circuits with varying lengh

### Section IV.B ###

- [ ] (@Simone-Bordoni) The claim “Convergence is reached after about 1.5e5 episodes” appears incorrect, as Figure 8 shows that fidelity continues to improve beyond this point.

I will modify this sentence as I think it was referred to a previous result.

- [ ] (@AlejandroSopena ) Elaboration on how “the average error on the trace distance introduced by measurement errors, shot noise, and additional gates” is computed?

consider using an apendix for more 'mathematical' results.

- [ ] (@Simone-Bordoni) Table II seems unnecessary, as the performance of RL and RB is comparable within one standard deviation.

I will remove the table, I agree with the reviewer.

- [ ] (@Simone-Bordoni , @AlejandroSopena) The manuscript mentions that RL requires fewer hardware resources than RB. This point is crucial, and the authors should quantify this advantage in order to achieve comparable performance as RB.

I think we should analyze better this point to give quantitative results (at least for our hardware implementation).

- [ ] (@BrunoLiegiBastonLiegi ) The classical resources required to train the RL agent should also be specified.

- [ ] (@BrunoLiegiBastonLiegi, @Simone-Bordoni ) Can the noise parameters be extracted from the RL agent? Are these parameters consistent across circuits or do they vary? A simple way to check this is to look at the action proposed by the RL agent. This should tell the authors whether the same kind of noise is inserted for the same gate at different points in a circuit.

We have already developed some code tom implement this function so it should not be difficult to retrieve some results

- [ ] (@Simone-Bordoni) When running on hardware, how much time needs to pass before the RL modelbecomes invalid?

I think a complete studio on this would take too long . I will add some considerations on this specifying that it mainly depends on the hardware setup.

- [ ] (@Simone-Bordoni, @AlejandroSopena, @BrunoLiegiBastonLiegi ) Is it possible for the authors to run the 3-qubit case on hardware as well? If not, what are the bottlenecks?

Let's see if we manage but I am a bit pessimistic if we want to finish in one month.

### Section V ###

- [ ] (@Simone-Bordoni) The claim that the lower fraction of two-qubit gates makes QFT and Grover’s circuits more challenging than the training set circuits should be elaborated upon. What structural differences make these circuits harder to model? An example of a training set circuit compared to QFT/Grover would be helpful. Consider combining Figures 10 and 11 and including a comparison circuit from the random training set.

I will add a bit more about this and consider adding a random circuit 

- [ ] (@Simone-Bordoni) The statement “This is one of the main reasons we chose the trace distance as the metric for the reward of the RL algorithm” needs clarification. This rationale should be introduced earlier, when the reward is first discussed.

I agree and I will move the sentence earlier

- [ ] (@Simone-Bordoni, @AlejandroSopena, @BrunoLiegiBastonLiegi ) The section on QFT and Grover’s circuits seems like a continuation of the evaluation of the RL agent on structured circuits. The authors should consider including this as a subsubsection of Section IV.A.

We should discuss this point, I don't agree with the reviewer but I can see his point, maybe we can try to give more reasons regarding the fact that this section is different from a standard evaluation

- [ ] (@Simone-Bordoni) The authors claim the RL agent can be used for circuit simulations, but how is this RL approach more efficient than simulating the exact noise model?

I think that he didn't understand that the RL approach should be used when the exact noise model is unknown...
The fact that he didn't understand this is a bit concerning.

### Section I ###

- [ ] (@Simone-Bordoni) The authors should provide more context on recent applications of reinforcement learning in quantum computing that “minimize heuristic assumptions” about the physical model. For example, relevant works in quantum control include:
a. Niu, M.Y., Boixo, S., Smelyanskiy, V.N. et al. Universal quantum control through deep reinforcement learning. npj Quantum Inf 5, 33 (2019).
b. Y. Baum, et. al. (2019) “Experimental Deep Reinforcement Learning for Error-Robust Gate-Set Design on a Superconducting Quantum Computer.” PRX Quantum 2(4), 040324.
c. Ho Nam Nguyen et al 2024 Mach. Learn.: Sci. Technol. 5 025066

I will add the 'relevant works' (He is probably looking to be cited...) also adding the proposals of the second reviewer

- [ ] (@Simone-Bordoni) The description of the Qibo framework and code availability could be moved to the end of the methodology section for a more logical flow.

### Section II.A ###

- [ ] (@AlejandroSopena ) In the statement “and an average error that is operation-independent, impacting only the qubits that are being manipulated”, did the authors mean to write “operation-dependent” instead?

- [ ] (@Simone-Bordoni  )  Can the author elaborate on which additionally tested noise channels that showed no significant performance enhancement?

I will add a few lines regarding the preliminary tests made with the thermal relaxation channel

- [ ] (@AlejandroSopena @Simone-Bordoni)The claim that “the parameters of these channels can be inferred either from calibration results or directly from the execution of quantum circuits, as demonstrated in this work” is problematic. At this point, the manuscript has not demonstrated this and later on, there is no discussion of parameter extraction.

Maybe we should rephrase and specify that we are fitting the model

### Section III.A ###

- [ ] (@Simone-Bordoni)  Typo: “native gates G_i_i=1^n, the initial...”

- [ ] (@Simone-Bordoni) The author should clarify that the entry for the CZ gate in the noise model matrix is set to 1 for the two qubit locations where the CZ gate is applied.

### Section IV.A ###

- [ ] (@Simone-Bordoni, @BrunoLiegiBastonLiegi ) In machine learning literature, datasets are typically divided into training, validation, and test sets. The validation set is for ensuring there is no overfitting and the test set is for evaluation on unseen data. The author should consider this naming convention.

While this convention is used in machine learning papers, for physics papers usually the opposite convention is used. Thus i preferred to use the 'physics' convention as this paper falls in this field. anyway it is fine for me to change.

- [ ] (@Simone-Bordoni, @AlejandroSopena, @BrunoLiegiBastonLiegi ) There are no significant differences in trends between the fidelity and trace distance. The authors should consider choosing one to make the results more concise.

Maybe also here we should vote the best preview

- [ ] (@Simone-Bordoni) Figure 4 - 9 should indicate that the shaded regions (no error bars) represent standard deviations.

### Section V ###

- [ ] (@Simone-Bordoni) Figure 10 introduces a gate labeled U_1 which should be defined.

- [ ] (@Simone-Bordoni) The terms “moment” and “depth” are used interchangeably throughout the manuscript. A single term should be used for consistency.

- [ ] (@Simone-Bordoni) For table IV, it is common practice to bold the best-performing numbers in each row of a results table. This would make it easier for readers to quickly identify the best-performing model.

## Reviewer 2 ##

- [ ] (@Simone-Bordoni) In addition, I think that a substantially extended literature review is necessary before the manuscript can be considered for publication in QST. The current version cites only 4 alternative works on noise learning and characterisation [31-34], however this is by now a huge field. Below is a (very) incomplete list of some alternative recent works on noise learning:
https://arxiv.org/abs/2502.03462
https://arxiv.org/abs/2501.06172
https://arxiv.org/abs/2303.00909
https://arxiv.org/abs/2206.06362
https://arxiv.org/abs/2311.15408
Indeed, I believe that to properly place this work into context, ideally an entire subsection should be dedicated to a thorough literature review, which also places this work into context, including the huge current effort to develop techniques for learning and characterising Lindbladians, which is currently neglected by the manuscript.

I will add the bibliography proposed and write an extra subsection for 'previous works'.
